[Tai Jun Jie] iP

TJun-Jie opened the pull request:
# Duke 

## Features
1. To do list
2. Search for your tasks
3. Support date types


It is **free**

- Text based
- easy to learn :satisfied:



```java
    public static void main(String[] args) {
        Duke duke = new Duke();
        duke.start();
    }
```

Task list
[ ] todo
[x] events
[ ] deadlines


> blockquote


[Google](https://www.google.com "Google's Homepage")


Inline `code`


 ~~Scratch this.~~

Commit:
Add Gradle support

TJun-Jie committed:
Level-1

TJun-Jie committed:
Level-2

TJun-Jie committed:
Level-3

TJun-Jie committed:
Level-4

TJun-Jie committed:
A-TextUiTesting

TJun-Jie committed:
Level-5

TJun-Jie committed:
Level-6

TJun-Jie committed:
A-Enums

TJun-Jie committed:
Update enums

TJun-Jie committed:
Add Storage Implementation

TJun-Jie committed:
Add logic for update and delete for Storage

TJun-Jie committed:
Add DateTime support for deadline and event tasks

TJun-Jie committed:
Fix Bug on split strings function

TJun-Jie committed:
Merge branch 'branch-Level-7'

TJun-Jie committed:
Merge branch 'branch-Level-8'

TJun-Jie committed:
Refactor code to cater to OOP principles

TJun-Jie committed:
Organize the classes into suitable java packages

TJun-Jie committed:
Merge branch 'add-gradle-support'

TJun-Jie committed:
Update build.gradle

TJun-Jie committed:
Add junit test

TJun-Jie committed:
Configure Jar file

TJun-Jie committed:
Add Documentation

TJun-Jie committed:
Merge branch 'branch-A-JavaDoc'

TJun-Jie committed:
Tweak code to follow convention

TJun-Jie committed:
Merge branch 'branch-A-CodingStandard'

TJun-Jie committed:
Add Find Function

TJun-Jie committed:
Merge branch 'branch-Level-9'

jorrdansoh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+                String name = Parser.getTodoName(command);
+                task = new Todo(name);
+
+            } else {
+
+                if (type.equals(TaskTypeEnum.deadline.toString())) {
+                    if (command.trim().length() == 8) {
+                        throw new EmptyNameException();
+                    }
+                    task = new Deadline(Parser.getDeadlineName(command), Parser.getDeadlineDate(command));
+                } else {
+                    if (command.trim().length() == 5) {
+                        throw new EmptyNameException();
+                    }
+                    task = new Event(Parser.getEventName(command), Parser.getEventDate(command));
+                }
+            }
```
The indentation here is quite deep, perhaps you can remove the nested `if` statements and check for `command.trim().length()` at the start?

jorrdansoh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,85 @@
+            int index = convertToInt(nextCommand);
+            storage.toggleDone(index, false);
+            taskList.setTaskAsUndone(index);
+        } else if (nextCommand.contains(CommandsEnum.todo.toString()) || nextCommand.contains(CommandsEnum.deadline.toString()) || nextCommand.contains(CommandsEnum.event.toString())) {
```
The `else if` condition is quite long, perhaps you can do a line wrap or declare a temporary variable to check?

jorrdansoh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.time.LocalDate;
+
+/**
+ * Represents the Deadline task which is a type of task
+ * It has the tag [D] and the time would be represented by "by 2022-10-23"
+ */
```
I like how you have JavaDoc comments for most of the public methods and classes :+1:

jorrdansoh commented on src/main/java/duke/CommandsEnum.java:
```diff
@@ -0,0 +1,5 @@
+package duke;
+
+public enum CommandsEnum {
+    bye, list, mark, unmark, delete, todo, deadline, event, find
```
I think `Enum`s should be written in all uppercase letters, since they are constants :) I noticed this in `TaskTypeEnum` too.

Kang-Quan replied:
Because they are constants, the names of an enum type's fields should be in uppercase letters :smiley:

jorrdansoh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,94 @@
+     * @return Date of event task
+     */
+    public static String getEventDate(String command) {
+        return command.split("/at", 2)[1].trim();
```
The `"/by"` and `"/at"` here seem like magic strings, perhaps you can declare static variables for them?

jorrdansoh commented on data/duke.txt:
```diff
@@ -0,0 +1,5 @@
+todo|0|test
+todo|0|test1
+todo|0|test 32423432
+todo|0|test34
```
Since people downloading the repo would probably want to have their own Duke data, perhaps you can `gitignore` your `duke.txt`?

jorrdansoh reviewed (commented):
Overall, I found your code easy to read for the most part except a few places
where the nesting was too deep. I noted a few minor coding standard violations
too. Some of the methods are getting quite long. Consider splitting into
smaller methods if that makes sense.

jorrdansoh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,183 @@
+    public void addLineToFile(String command) {
+        try {
+            String[] combiStr;
+            String[] split = command.split(" ", 2);
```
The name `split` here is the same as the method `split`, perhaps you can change it to something like `splitString`?

Kang-Quan replied:
I have noticed the variable name `split` and the method `split` has the same name. To prevent confusion I suggest changing the variable name to `splitCommand`.

jorrdansoh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+    /**
+     * Displays all tasks in the command line
+     */
+    public void displayAllTask() {
```
Since you're displaying all the `Task`s with this method, perhaps you can name it `displayAllTasks` instead?

TJun-Jie committed:
Add GUI support

TJun-Jie committed:
Merge branch 'branch-Level-10'

Kang-Quan reviewed (commented):
Overall, I found your code easy to read for the most part. Most public classes have JavaDocs comments making it easier to understand. However, I noted a few minor coding standard violations. Some conditional statements were too long, consider abstracting them out into smaller methods if that make sense.

Kang-Quan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.time.LocalDate;
+
+/**
```
I like how most public method and classes have JavaDocs comments making it easier to read your code. 👍

Kang-Quan commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    @Override
+    public String toString() {
+        String tag = "[E]";
```
I noticed there is a magic String, "[E]". Perhaps, I suggest declaring a static variable for it. I have noticed this in Todo and Deadline classes too.

Kang-Quan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,117 @@
+            int index = convertToInt(nextCommand);
+            storage.toggleDone(index, false);
+            return taskList.setTaskAsUndone(index);
+        } else if (nextCommand.contains(CommandsEnum.todo.toString()) || nextCommand.contains(CommandsEnum.deadline.toString()) || nextCommand.contains(CommandsEnum.event.toString())) {
```
I noticed the condition in the `else if` statement has exceeded the hard limit of 120 characters. I have a few suggestions, firstly to use line wrap at appropriate place, abstract out the condition in a Boolean variable or abstracting this big method into smaller methods. I noticed this issue in other `else if` statements in this methods.

Kang-Quan reviewed (commented):
Overall, I found your code easy to read for the most part. Most public classes have JavaDocs comments making it easier to understand. However, I noted a few minor coding standard violations. Some conditional statements were too long, consider abstracting them out into smaller methods if that make sense.

TJun-Jie committed:
Update build.gradle

Change build.gradle to configure jar file to open GUI application

Change main class name in application from Duke to Laucher

TJun-Jie committed:
Add assertions

Use of Assertions to document important assumptions that should hold at various points in the code.

TJun-Jie committed:
A-CodeQuality

Improve code quality by following conventions

TJun-Jie committed:
Merge pull request #2 from TJun-Jie/branch-A-CodeQuality

A-CodeQuality

TJun-Jie committed:
Merge branch 'branch-A-Assertion'

TJun-Jie committed:
Update Ui.java

TJun-Jie committed:
Merge branch 'branch-A-Assertions'

TJun-Jie committed:
Add Tag support

Added tagging support for task so that the task can be separated into categories.

TJun-Jie committed:
Merge branch 'branch-tagging'

TJun-Jie committed:
Add tag functionality

Added showTag command which shows all the tasks under that tag

TJun-Jie committed:
Configure checkstyle and refactor code

TJun-Jie committed:
Add Ui Picture

TJun-Jie committed:
Update User Guide

TJun-Jie committed:
Update user guide

TJun-Jie committed:
Fix bug in storage class

TJun-Jie committed:
Merge branch 'master' of https://github.com/TJun-Jie/ip

TJun-Jie committed:
Update README.md

TJun-Jie committed:
Update README.md
